wordpress-mobile/WordPress-Android
# [Comments section] Update the comment list

## Fixes #15397

An automattician reported the comment list did not update after comment editing. Opening the comment details from the list and editing the comment section didn't reflect the changes on the main list, so it is needed a PTR on the list to make the changes visible.

This PR is going to update the main comment list when the user comes back from the comment detail screen.

As suggested @khaykov in the comment section, I'm going to:

- [x] Leave `localCommentCacheUpdateHandler` as a constructor parameter
- [x] Remove the delay, use only `performSingleCommentModeration` and understand better its code
- [x] Change the `SharedPreference` strategy
- [x] Add unit testing to test the new changes

## To test:

- Go in My Site > Comments in a site with at least one comment
- Tap on a comment to open the comment details view
- From the overflow menu select Edit, make some changes and save them
- Notice you are landed in comment details and that is updated
- Go back to the comments list and notice the original comment entry in the list is correctly updated

## Regression Notes
1. Potential unintended areas of impact:
When the comment is moderated and edited at the same time, the main list could not refresh the updated comment.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manual testing

3. What automated tests I added (or what prevented me from doing so)
I have updated the test `onActionMenuClicked triggers DONE action if comment update successfully` to verify if the `requestCommentsUpdate` method is called from `localCommentCacheUpdateHandler`

## PR submission checklist:
